src/org/omegat/filters2/text/ini/INIFilter.java

```diff
@@ -107,37 +122,47 @@ public void processFile(BufferedReader reader, BufferedWriter outfile, FilterCon
                 continue;
             }
 
+            // retrieve section as group name
             if (trimmed.startsWith("[") && trimmed.endsWith("]")) {
-                // group name
                 group = trimmed.substring(trimmed.offsetByCodePoints(0, 1),
                         trimmed.offsetByCodePoints(trimmed.length(), -1));
+                outfile.write(str);
+                outfile.write(lbpr.getLinebreak());
+                continue;
             }
 
             // key=value pairs
-            int equalsPos = str.indexOf('=');
+            equalsPos = str.indexOf('=');
 
-            // if there's no separator, assume it's a key w/o a value
+            // if there's no separator, assume it's an unnamed key with a value
+            // It may be a continuous line, so allow translate.
             if (equalsPos == -1) {
-                equalsPos = str.offsetByCodePoints(str.length(), -1);
-            }
-
-            // advance if there're spaces after =
-            while (str.codePointCount(equalsPos, str.length()) > 1) {
-                int nextOffset = str.offsetByCodePoints(equalsPos, 1);
-                if (str.codePointAt(nextOffset) != ' ') {
-                    break;
+                // produce virtual key
+                key = buildKey(group, unnamedCounter);
+                unnamedCounter++;
+                // advance if there are spaces before contents
+                afterEqualsPos = 0;
+                while (str.codePointCount(afterEqualsPos, str.length()) > 1) {
+                    if (str.codePointAt(afterEqualsPos) != ' ') {
+                        break;
+                    }
+                    afterEqualsPos = str.offsetByCodePoints(afterEqualsPos, 1);
+                }
+            } else {
+                // advance if there're spaces after =
+                while (str.codePointCount(equalsPos, str.length()) > 1) {
+                    int nextOffset = str.offsetByCodePoints(equalsPos, 1);
+                    if (str.codePointAt(nextOffset) != ' ') {
+                        break;
+                    }
+                    equalsPos = nextOffset;
                 }
-                equalsPos = nextOffset;
+                key = (group != null ? group + '/' : "") + str.substring(0, equalsPos).trim();
```

Comment:
I think this would also be clearer as a `build` method, but it's OK if you don't want to bother.

(I know this is from the existing code.)

Reply:
Use buildKey() for here now